Automattic/woocommerce-payments
# Add Transaction Details notice when a dispute has staged evidence

Fixes #6966 


> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`


#### Changes proposed in this Pull Request









This PR adds a notice to the Transaction Details screen when the dispute associated with the transaction has staged evidence (i.e. the merchant has clicked "Save for later" on the dispute challenge form). This notice reminds the merchant that they have a draft response to the dispute.

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/99dceaf3-6fa5-4450-8960-5dd221d73b3c)

I've also added a test suite for `DisputeDetails`, a copy of the test suite implemented in the WIP PR #7093.

#### Testing instructions





* Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
* Create a disputed transaction using the card `4000000000000259` at checkout.
* Find the dispute in Payments → Disputes and click it to view the dispute details screen.
* Click on the Transaction ID to view the Transaction Details screen.
* You should **not** see the staged dispute evidence notice.
* Navigate to the dispute details screen again.
* Click **Challenge**.
* Make a small change to this form and click "Save for Later" to stage the dispute evidence submission.
* Navigate back to the Transaction Details screen.
* You **should now** see the staged dispute evidence notice.

-------------------


**Post merge**
